ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/views/PaywallView.kt

```diff
@@ -89,11 +89,26 @@ class PaywallView : FrameLayout {
      */
     fun setOfferingId(offeringId: String?) {
         this.offeringId = offeringId
+        recreateComposeView()
         invalidate()
     }
 
     private fun init(context: Context, attrs: AttributeSet?) {
         parseAttributes(context, attrs)
+        recreateComposeView()
```

Comment:
This is difficult to review, but I've basically extracted adding the compose view to a function